### PR TITLE
all: Add tls.cipher-suites config option

### LINKED ISSUE #3970: Option to configure the cipher suites
#### Summary
An option to configure the cipher suites to be allowed

#### Why do we need this?
To allow only certain cipher suites for improved security

#### What is already there? What do you see now?
- The default list of cipher suites specified in Go runtime is used

#### What is missing? What do you want to see?
- An option to configure certain cipher suites from the default list

#### Environment
TTS v3.11.3

#### How do you propose to implement this?
- Add `tls.cipher-suites` option 

#### How do you propose to test this?
...
#### Can you do this yourself and submit a Pull Request?
No

### PR DESCRIPTION
**Pull request by michalborkowski96**
#### Summary
Closes #3970 

#### Changes
Added an option to configure used cipher suites

#### Testing
Local tests using `openssl s_client -cipher "$cipher" -connect`

##### Regressions
None

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/component/tls.go [hunk 1]
```diff
@@ -58,9 +58,13 @@ func WithNextProtos(protos ...string) TLSConfigOption {
 // GetTLSServerConfig gets the component's server TLS config and applies the given options.
 func (c *Component) GetTLSServerConfig(ctx context.Context, opts ...TLSConfigOption) (*tls.Config, error) {
 	conf := c.GetBaseConfig(ctx).TLS
+	cipherSuites, err := conf.GetCipherSuites()
+	if err != nil {
+		return nil, err
+	}
 	res := &tls.Config{
-		MinVersion:               tls.VersionTLS12,
-		PreferServerCipherSuites: true,
```
Comment (michalborkowski96) [inline pkg/component/tls.go:63]: `PreferServerCipherSuites` has been removed because this field is deprecated and ignored

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ For details about compatibility between different releases, see the **Commitment
 
 ### Added
 
+- `tls.cipher-suites` config option to specify used cipher suites
```
Comment (adriansmares) [inline CHANGELOG.md]: Nit: `.` at the end

File: pkg/component/tls.go [hunk 3]
```diff
@@ -58,9 +58,13 @@ func WithNextProtos(protos ...string) TLSConfigOption {
 // GetTLSServerConfig gets the component's server TLS config and applies the given options.
 func (c *Component) GetTLSServerConfig(ctx context.Context, opts ...TLSConfigOption) (*tls.Config, error) {
 	conf := c.GetBaseConfig(ctx).TLS
+	cipherSuites, err := conf.GetCipherSuites()
+	if err != nil {
+		return nil, err
+	}
```
Comment (adriansmares) [inline pkg/component/tls.go:69]: If the list is not empty, print a warning such as 
`TLS is configured to use a custom set of cipher suites. Make sure that your list is up to date, or disable the custom cipher suites`.

We want to make sure that people don't shotgun themselves in the feet.

**Review by adriansmares — APPROVED** [review]

File: pkg/config/tlsconfig/config.go [hunk 4]
```diff
@@ -169,20 +169,43 @@ func readCert(fileReader FileReader, certFile, keyFile string) (*tls.Certificate
...
-	Key         string     `json:"key" yaml:"key" name:"key" description:"Location of TLS private key"`
-	ACME        ACME       `name:"acme"`
-	KeyVault    KeyVault   `name:"key-vault"`
+	Source       string     `name:"source" description:"Source of the TLS certificate (file, acme, key-vault)"`
+	FileReader   FileReader `json:"-" yaml:"-" name:"-"`
+	Certificate  string     `json:"certificate" yaml:"certificate" name:"certificate" description:"Location of TLS certificate"`
+	Key          string     `json:"key" yaml:"key" name:"key" description:"Location of TLS private key"`
+	ACME         ACME       `name:"acme"`
+	KeyVault     KeyVault   `name:"key-vault"`
+	CipherSuites []string   `name:"cipher-suites" description:"List of IANA names of TLS cipher suites to use"`
```
Comment (adriansmares) [inline pkg/config/tlsconfig/config.go]: Also prefix the description with `DEPRECATED: `, as this is a function that shouldn't be used unless you're very sure what you're doing.

**Review by htdvisser — APPROVED** [review]
Nothing to add

**Pull request merged**